File: src/python/utils/logger.py
import logging
import sys
from typing import Optional

def setup_logger(verbose: bool = False, name: Optional[str] = None) -> logging.Logger:
    """Setup logger with appropriate level and formatting"""
    logger = logging.getLogger(name or 'rhythmic')
    
    # Prevent duplicate handlers
    if logger.handlers:
        return logger
    
    # Set level
    level = logging.DEBUG if verbose else logging.INFO
    logger.setLevel(level)
    
    # Create formatter
    if verbose:
        formatter = logging.Formatter(
            '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
        )
    else:
        formatter = logging.Formatter('%(message)s')
    
    # Create console handler
    handler = logging.StreamHandler(sys.stdout)
    handler.setLevel(level)
    handler.setFormatter(formatter)
    
    logger.addHandler(handler)
    
    return logger

File: src/js/utils/logger.js
import chalk from 'chalk';

export class Logger {
  constructor(verbose = false) {
    this.verbose = verbose;
  }
  
  info(message) {
    console.log(message);
  }
  
  success(message) {
    console.log(chalk.green(message));
  }
  
  warning(message) {
    console.log(chalk.yellow(message));
  }
  
  error(message) {
    console.error(chalk.red(message));
  }
  
  debug(message) {
    if (this.verbose) {
      console.log(chalk.gray(message));
    }
  }
  
  step(message) {
    console.log(chalk.blue(message));
  }
}